src/WebJobs.Extensions.DurableTask/DurableTaskJobHostConfigurationExtensions.cs

```diff
@@ -27,7 +26,8 @@ public static IWebJobsBuilder AddDurableTask(this IWebJobsBuilder builder)
 
             builder.AddExtension<DurableTaskExtension>()
                 .BindOptions<DurableTaskOptions>()
-                .Services.AddSingleton<IConnectionStringResolver, WebJobsConnectionStringProvider>();
+                .Services.AddSingleton<IConnectionStringResolver, WebJobsConnectionStringProvider>()
+                         .AddSingleton<IOrchestrationServiceFactory, OrchestrationServiceFactory>();
```

Comment:
With this model, do you think it would be possible for customers to provide their own `IOrchestrationServiceFactory` in the future via our DI support?

Reply:
I'm not quite sure what our customer facing DI support will look like, and especially how they will interact with extensions. I don't see why this approach wouldn't allow customers to provide their own `IOrchestrationServiceFactory` though.